src/deepsparse/server/server.py

```diff
@@ -0,0 +1,159 @@
+        num_streams=server_config.num_workers,
+        scheduler=Scheduler.multi_stream,
+    )
+
+    _LOGGER.info(f"Built context: {repr(context)}")
+
+    app = FastAPI()
+
+    @app.get("/", include_in_schema=False)
+    def _home():
+        return RedirectResponse("/docs")
+
+    @app.get("/config", tags=["general"], response_model=ServerConfig)
+    def _info():
+        return server_config
+
+    @app.get("/ping", tags=["general"], response_model=bool)
+    @app.get("/health", tags=["general"], response_model=bool)
+    @app.get("/healthcheck", tags=["general"], response_model=bool)
+    @app.get("/status", tags=["general"], response_model=bool)
+    def _health():
+        return True
+
+    app._deepsparse_pool = ThreadPoolExecutor(max_workers=context.num_streams)
+
+    add_invocations_endpoint = len(server_config.endpoints) == 1
+
+    # fill in names if there are none
+    for idx, endpoint_config in enumerate(server_config.endpoints):
+        if endpoint_config.name is None:
+            endpoint_config.name = f"endpoint-{idx}"
+
+    # creat pipelines & endpoints
+    for endpoint_config in server_config.endpoints:
+        pipeline_config = endpoint_config.to_pipeline_config()
+        pipeline_config.kwargs["executor"] = app._deepsparse_pool
+
+        _LOGGER.info(f"Initializing pipeline for '{endpoint_config.name}'")
+        pipeline = Pipeline.from_config(pipeline_config, context)
+
+        _LOGGER.info(f"Adding endpoints for '{endpoint_config.name}'")
+        _add_pipeline_endpoint(app, endpoint_config, pipeline, add_invocations_endpoint)
+
+    _LOGGER.info(f"Added endpoints: {[route.path for route in app.routes]}")
+
+    return app
+
+
+def _add_pipeline_endpoint(
+    app: FastAPI,
+    endpoint_config: EndpointConfig,
+    pipeline: Pipeline,
+    add_invocations_endpoint: bool = False,
+):
+    input_schema = pipeline.input_schema
+    output_schema = pipeline.output_schema
+
+    pool: ThreadPoolExecutor = app._deepsparse_pool
+
+    route = endpoint_config.endpoint or "/predict"
+    if not route.startswith("/"):
+        route = "/" + route
+
+    @app.post(route, tags=["predict"], response_model=output_schema)
+    async def _predict_func(request: pipeline.input_schema):
+        return await get_running_loop().run_in_executor(pool, pipeline, request)
+
+    _LOGGER.info(f"Added '{route}' endpoint")
+
+    if hasattr(input_schema, "from_files"):
+        file_route = route + "/files"
+
+        @app.post(file_route, tags=["predict"], response_model=output_schema)
+        async def _predict_from_files_func(request: List[UploadFile]):
+            request = pipeline.input_schema.from_files(
+                (file.file for file in request), from_server=True
+            )
+            return await get_running_loop().run_in_executor(pool, pipeline, request)
+
+        _LOGGER.info(f"Added '{file_route}' endpoint")
+
+    if add_invocations_endpoint and hasattr(input_schema, "from_files"):
```

Comment:
we need `invocations` for situations besides from_files, such as nlp which doesn't currently support from_files

Reply:
Gotcha, I was translating the logic from the existing server, which only allows file uploads with sagemaker. You okay with doing sagemaker cleanup in a separate PR?

Reply:
surprising, our advertised examples use the nlp pipelines which don't have file support IIRC, maybe this was a late introduction post the last time we tested sagemaker

Reply:
Ahh I was interpretting code incorrectly - nevermind!